modal/app.py

```diff
@@ -550,6 +553,24 @@ def wrapped(
                 webhook_config = None
                 raw_f = f
 
+                if not is_top_level_function(f) and not serialized:
+                    raise InvalidError(
+                        dedent(
+                            """
+                            The `@app.function` decorator cannot be used on class methods.
+                            Please use `@app.cls` with `@modal.method` instead. Example:
+
+                            ```python
+                            @app.cls()
+                            class MyClass:
+                                @method()
```

Comment:
```suggestion
                                @modal.method()
```

to be more consistent with the prose, and because I think this is a better default style for our docs.

Reply:
I think the docs mostly use `@method()` rather than `@modal.method`. No super strong feelings but I think consistency favors the former?

Reply:
By consistency I meant that the text of the error message has

> Please use `@app.cls` with `@modal.method` instead.

I agree we're not very consistent with the docs, and personally would prefer always using the full `modal.<function>` spelling. Otherwise it is annoying because you need to update both the code where the error was raised and also change your imports. Additionally, since we use such general names, we run the risk of cascading confusing errors if the user happened to have a variable named `method` or whatever in their script (this is riskiest with `exit`, which is a built-in function).

Note that the ambiguity would be eliminated if the partial function decorators were methods on `App` instead of standalone functions in the `modal` namespace :)